app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -2,126 +2,97 @@ module ManageIQ::Providers
   class Hawkular::MiddlewareManager::LiveMetricsCapture
     class MetricValidationError < RuntimeError; end
 
+    ONE_HOUR = 60 * 60
+    ONE_DAY = 60 * 60 * 24
+    ONE_WEEK = ONE_DAY * 7
+
     def initialize(target)
       @target = target
       @ems = @target.ext_management_system
-      @metrics_client = @ems.metrics_client
-      @gauges = @metrics_client.gauges
-      @counters = @metrics_client.counters
-      @avail = @metrics_client.avail
+      @prometheus_client = @ems.prometheus_client
       @included_children = @target.included_children
       @supported_metrics = @target.supported_metrics
     end
 
     def fetch_metrics_available
       resource = @included_children ? @ems.resource_tree(@target.ems_ref) : @ems.resource(@target.ems_ref)
       resource.metrics(!!@included_children)
-              .select { |m| @supported_metrics.key?(m.hawkular_type_id) }.collect do |metric|
-        parse_metric(metric)
-      end
-    end
-
-    def parse_metric(metric)
-      {:id => metric.hawkular_id, :name => @supported_metrics[metric.hawkular_type_id], :type => metric.hawkular_type, :unit => metric.unit}
-    end
-
-    def parse_metrics_ids(metrics)
-      gauge_ids = []
-      counter_ids = []
-      avail_ids = []
-      metrics_ids_map = {}
-      metrics.each do |metric|
-        metric_id = metric[:id]
-        case metric[:type]
-        when "GAUGE"        then gauge_ids.push(metric_id)
-        when "COUNTER"      then counter_ids.push(metric_id)
-        when "AVAILABILITY" then avail_ids.push(metric_id)
-        end
-        metrics_ids_map[metric_id] = metric[:name]
-      end
-      [gauge_ids, counter_ids, avail_ids, metrics_ids_map]
+              .select { |metric| @supported_metrics[@target.live_metrics_type].key?(metric.name) }
+              .collect do |metric|
+                metric_hash = metric.to_h
+                metric_hash['name'] = @supported_metrics[@target.live_metrics_type][metric.name]
+                metric_hash
+              end
+              .to_a
```

Comment:
Is `to_a` redundant here?
Asking because I don't know the type of the collection. Looks like it is already an array.

Reply:
Point is that collect returns a collection, and it doesn't have access to the .first method used.
So, I convert it using .to_a for that reason.